omnibus/files/private-chef-ctl-commands/open_source_chef12_upgrade.rb

```diff
@@ -359,21 +359,33 @@ def make_dir(dir, permissions)
     Dir.mkdir(dir, permissions) unless File.directory?(dir)
   end
 
-  # TODO(jmink) Add error handling
+  # Any config not specified on the command line
+  # should be pulled from /etc/chef-server/chef-server-running.json
   def pull_chef11_db_credentials
-    # This code pulled from knife-ec-backup and adapted
-    log "Pulling open source Chef 11 database credentials"
-    if !File.exists?("/etc/chef-server/chef-server-running.json")
-      log "Failed to find /etc/chef-server/chef-server-running.json"
-      exit 1
-    end
+    sql_host = @options.sql_host
+    sql_port = @options.sql_port
+    sql_pass = @options.sql_pass
+    sql_user = @options.sql_user
 
-    running_config = JSON.parse(File.read("/etc/chef-server/chef-server-running.json"))
-    sql_host = running_config['chef_server']['postgresql']['vip']
-    sql_port = running_config['chef_server']['postgresql']['port']
-    sql_user = running_config['chef_server']['postgresql']['sql_user']
-    sql_password = running_config['chef_server']['postgresql']['sql_password']
-    [sql_host, sql_port, sql_user, sql_password]
+    if !(sql_host && sql_port && sql_pass && sql_user)
+      log "Reading SQL configuration from /etc/chef-server/chef-server-running.json"
+      begin
+        running_config = JSON.parse(File.read("/etc/chef-server/chef-server-running.json"))
+        sql_host ||= running_config['chef_server']['postgresql']['vip']
```

Comment:
What happens if I leave off one of the sql_* options but specify 2 or 3 on the command line? If my reading of the if statement is right, we enter the block to set the options from the running.json. But with the ||= we won't over write any that were set on the command line, so we'll end up with mixed options. This might work, or fail spectacularly.

It seems like we should make this all or nothing. You set all of them, or we pull all of them from running.json.